src/EnergyPlus/AirTerminalUnit.hh

```diff
@@ -0,0 +1,118 @@
+#ifndef AirTerminalUnit_hh_INCLUDED
+#define AirTerminalUnit_hh_INCLUDED
+
+// C++ Headers
+#include <string>
+
+// EnergyPlus Headers
+#include <EnergyPlus.hh>
+
+namespace EnergyPlus {
+
+// types of air terminal units, refactored from old DataDefineEquip
+enum AirTerminalUnitType {
+	notYetDetermined,
+	dualDuctConstVolume,
+	dualDuctVAV,
+	singleDuctVAVReheat,
+	singleDuctConstVolReheat,
+	singleDuctVAVNoReheat,
+	singleDuct_SeriesPIU_Reheat,
+	singleDuct_ParallelPIU_Reheat,
+	singleDuct_ConstVol_4PipeInduc,
+	singleDuctVAVReheatVSFan,
+	singleDuctCBVAVReheat,
+	singleDuctCBVAVNoReheat,
+	singleDuctConstVolCooledBeam,
+	dualDuctVAVOutdoorAir,
+	singleDuctUserDefined,
+	singleDuctInletATMixer,
+	singleDuctSupplyATMixer,
+	singleDuctConstVolFourPipeBeam
+};
+
+// base class for all air distribution units.  zone air terminals for connecting to central air handlers
+class AirTerminalUnit
+{
+
+protected: // Creation
+
+	// Default Constructor
+	AirTerminalUnit() :
+		terminalType( notYetDetermined ),
+		aDUNum( 0 ),
+		airAvailSchedNum( 0 ),
+		airAvailable( false ),
+		vDotDesignPrimAir( 0.0 ),
+		vDotDesignPrimAirWasAutosized( false ),
+		mDotDesignPrimAir( 0.0 ),
+		airInNodeNum( 0 ),
+		airOutNodeNum( 0 ),
+		zoneIndex( 0 ),
+		zoneNodeIndex( 0 )
+	{}
+
+	// Copy Constructor
+	AirTerminalUnit( AirTerminalUnit const & ) = default;
+
+	// Move Constructor
+#if !defined(_MSC_VER) || defined(__INTEL_COMPILER) || (_MSC_VER>=1900)
+	AirTerminalUnit( AirTerminalUnit && ) = default;
+#endif
+
+public: // Creation
+
+	// Destructor
+	virtual
+	~AirTerminalUnit()
+	{}
+
+protected: // Assignment
+
+	// Copy Assignment
+
+	AirTerminalUnit &
+	operator =( AirTerminalUnit const & ) = default;
+
+	// Move Assignment
+#if !defined(_MSC_VER) || defined(__INTEL_COMPILER) || (_MSC_VER>=1900)
+	AirTerminalUnit &
+	operator =( AirTerminalUnit && ) = default;
+#endif
+
+public: // Methods		REMOVE ANY OF THESE THAT AREN'T COMMON (WITH SAME ARGS) TO ALL SUB-TYPES
+
+	// for unit tests
+	virtual
+	void
+	clear_state() = 0;
+
+	virtual
+	void
+	simulate(
+		bool const FirstHVACIteration, // TRUE if first HVAC iteration in time step
+		Real64 & NonAirSysOutput // convective cooling by the beam system [W]
+	) = 0;
+
+
+
+public: // Data
```

Comment:
@EnergyArchmage Yeah in the base class it would have to be protected if a derived class is using that variable. Basically, private makes it so ONLY that class has access while protected allows that classes and any derived classes access to that data. 

Lets try to pull some more people and their thoughts in on this... @Myoldmopar @kbenne